### ISSUE TITLE
bugs introduced recently for ksampler, and control_after_generate=fixed

### ISSUE THREAD

**Issue by ppbrown** [comfyanonymous/ComfyUI#2349]
old behaviour(good):

if on random, and you generate an image, and you like it.. you can cutnpaste the seed and share it. Or just change control-after-generate to be fixed, and it will stay generating with that same seed you liked.

Additionally, if you were on fixed, then you changed it to random, then if you hit the Queue button, it would automatically generate a new seed and a new render.

current behaviour(bad):

If on random, and you wanted to save the random seed used in the ksampler that generated the latent you liked....
You would have had to save it, BEFORE you hit the Queue button.
It seems like now, somehow, AS SOON AS you hit Queue, it generates a new seed, and displays it, before the new latent has even finished generating

Additionally, if you are on fixed, then you change it to random, you can Queue, and it will do seemingly nothing.
You have to hit it a second time to then have it pick a new random seed and use it.

**Comment by suede299** [comment]
Use [ComfyUI-Inspire-Pack](https://github.com/ltdrdata/ComfyUI-Inspire-Pack)'s global seed node, I guess. If you adapted other UI's to display the current image used seed.

**Comment by ppbrown** [comment]
um, what?
I did nothing. I didnt even add any custom nodes.
I'm using a pure comfuyi git pull, as managed by stableswarmui.

Are you saying this is  NOT broken for a non-stablilityswarm integrated comfyui?  that would be very unexpected.


Ive been using the "before" behaviour for months. Its the same as the stabilityswarmui behaviour for the random/notrandom toggle.

But now, the comfyui on the backend is messed up with this, but stabilityswarmui is not.
its really wierd.

**Comment by suede299** [comment]
> um, what? I did nothing. I didnt even add any custom nodes. I'm using a pure comfuyi git pull, as managed by stableswarmui.
> 
> Are you saying this is NOT broken for a non-stablilityswarm integrated comfyui? that would be very unexpected.
> 
> Ive been using the "before" behaviour for months. Its the same as the stabilityswarmui behaviour for the random/notrandom toggle.
> 
> But now, the comfyui on the backend is messed up with this, but stabilityswarmui is not. its really wierd.
That's me misinterpreting your question. I'm also using StabilitySwarmUI like you, but for the way the seed is displayed, it never shows the image as the next seed value as soon as the image starts to be generated, and when it's set to fix, it will generate the image using the currently displayed seed value (it's really just not generating and displaying the seed value for the next time it's used).
So in order to see the seed value used for the image that is currently finished generating, I used this custom node.

![image](https://github.com/comfyanonymous/ComfyUI/assets/85022330/976f363f-2d4b-4419-86a7-3599f263eb8b)

**Comment by ppbrown** [comment]
For the record, i went through the hassle of pulling  a brand new pristine standalone comfyui install.
Bug is present

**Comment by comfyanonymous** [comment]
It has always been like the "current behaviour" you probably had an addon that changes it.

**Comment by ppbrown** [comment]
nope, no addons. no extra nodes, nothing

**Comment by ppbrown** [comment]
that being said... even if Im wrong about past behaviour,   I dont see how the "current behaviour" makes sense.
Could you please change it to bring it in line with stabilityswarm?

**Comment by Jordain** [comment]
Yeah I have the same issue. If I switch from fixed to randomize and generate it still keeps the same image. If I switch from randomize to fixed it will generate a new seed, until I generate again. I copied the previous seed from the random output and put it in the generate on the fixed and it worked. Looks like Ksampler doesn't  actually apply the control_after_generate until after you generate it once.

**Comment by K-Zero** [comment]
This issue is still present. It seems some people think this is a feature including the devs because there is a option to change that in the settings.
**Settings -> Node Widget -> Widget control mode -> Set to "before"**
This should be set as default just like in any other application in my opinion.
